File: scripts/ask-liquidsoap.pl
#!/usr/bin/perl -w

use strict ;
use Net::Telnet ;

my $telnet = new Net::Telnet ( Timeout=>10, Errmode=>'die', Port=>1234) ;
$telnet->open('localhost') ;

die "Usage: $0 <command>\n" unless @ARGV ;
$telnet->print($ARGV[0]) ;
my ($output,$end) = $telnet->waitfor('/END$/') ;
print $output;

File: scripts/ask-liquidsoap.rb
#!/usr/bin/ruby

require 'net/telnet'

liq_host = "localhost"
liq_port = 1234

conn = Net::Telnet::new("Host" => liq_host, "Port" => liq_port)

conn.puts(ARGV[0])
conn.waitfor("Match" => /^END$/) do |data|
  puts data.sub(/\nEND\n/,"")
end
